delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storages/storage/StoragesConfig.scala

```diff
@@ -0,0 +1,116 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storages.storage
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.plugins.storages.config.{AggregateConfig, IndexingConfig}
+import ch.epfl.bluebrain.nexus.delta.plugins.storages.storage.StoragesConfig.StorageTypeConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.storages.storage.model.DigestAlgorithm
+import ch.epfl.bluebrain.nexus.delta.sdk.cache.KeyValueStoreConfig
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.AuthToken
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.PaginationConfig
+
+import java.nio.file.Path
+
+/**
+  * Configuration for the Storages module.
+  *
+  * @param aggregate         configuration of the underlying aggregate
+  * @param keyValueStore     configuration of the underlying key/value store
+  * @param pagination        configuration for how pagination should behave in listing operations
+  * @param indexing          configuration of the indexing process
+  * @param storageTypeConfig configuration of each of the storage types
+  */
+final case class StoragesConfig(
+    aggregate: AggregateConfig,
+    keyValueStore: KeyValueStoreConfig,
+    pagination: PaginationConfig,
+    indexing: IndexingConfig,
+    storageTypeConfig: StorageTypeConfig
+)
+
+object StoragesConfig {
+
+  /**
+    * The configuration of each of the storage types
+    * @param disk          configuration for the disk storage
+    * @param amazon        configuration for the s3 compatible storage
+    * @param remoteDisk    configuration for the remote disk storage
+    */
+  @SuppressWarnings(Array("OptionGet"))
+  final case class StorageTypeConfig(
+      disk: DiskStorageConfig,
+      amazon: Option[S3StorageConfig],
+      remoteDisk: Option[RemoteDiskStorageConfig]
+  ) {
+    lazy val amazonUnsafe: S3StorageConfig             = amazon.get
```

Comment:
this is used in the code, so it can throw an error... if that is the case, we should maybe use default values or handle the error downstream

Reply:
If you see the state machine, it is used in a way that it can't throw.
Besides that, the name says it is unsafe, which tells you not to use it unless you know what you're doing.